boot/boot_serial/src/boot_serial.c

```diff
@@ -78,7 +78,12 @@
 
 BOOT_LOG_MODULE_DECLARE(mcuboot);
 
+#if defined(CONFIG_BOOT_MAX_LINE_INPUT_LEN) && CONFIG_BOOT_MAX_LINE_INPUT_LEN > 512
```

Comment:
I would avoid adding `CONFIG_` things here.
I would rather have `MCUBOOT_` identifier defined in mcuboot_config.h and used here instead of the `CONFIG_`.
So the same logic here but `MCUBOOT_` used.